src/components/views/rooms/RoomHeader.js

```diff
@@ -320,10 +323,15 @@ module.exports = React.createClass({
         }
 
         let rightRow;
+
         if (!this.props.editing) {
             rightRow =
                 <div className="mx_RoomHeader_rightRow">
                     { settingsButton }
+                    <ManageIntegsButton
+                        onCancelClick={this.props.onCancelClick}
```

Comment:
It seems _quite_ bizarre to be bubbling the `onCancelClick` of the integrations manager to the `onCancelClick` of the room header. Stranger still that the `onCancelClick` that is passed to RoomView will close room settings, so this is definitely wrong.